### PR TITLE
Refactor `with_channel_timeout`

### PR DESCRIPTION
**Pull request by coszio**
**Note**: This should only affect the inter-node channels.

- Removes the intermediate timeout implementation that was working separately from the request and channel timeouts.
- Now the channel timeout is set to a hard 1 hour (not configurable)
- All the individual requests now pass through an interceptor, which checks if they already have a timeout set, and if not, they set it to the `grpc_timeout_ms` from settings.

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [x] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [x] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by timvisee — APPROVED** [review]
Nice! I tested this together with <https://github.com/qdrant/qdrant/pull/2748> on a two node cluster and it seems to work well.

One thing we should be aware about is that if an internal call times out, it is retried for up to two times. Imagine that you set a timeout of 3 seconds, the total request may take 9 seconds (due to transparent retry). I think this is correct behavior, but I wanted to mention it anyway.

**Comment by coszio** [comment]
A nice to have would be to improve the error message. Ideally mentioning "timeout error after *n* ms" instead of just "operation cancelled"

**Comment by timvisee** [comment]
> A nice to have would be to improve the error message. Ideally mentioning "timeout error after _n_ ms" instead of just "operation cancelled"

Or possibly even "(...) after _n_ seconds and _x_ retries"

**Comment by coszio** [comment]
Looks like timeout ms is not easily accessible at the point of conversion, as it should come from `tonic` crate and it is not implemented in that way. I only improved in a way that the "Timeout exceeded" message is actually shown. 

For some reason [tonic returns `Status::Cancelled` instead of `Status::DeadlineExceeded` for timeouts](https://github.com/hyperium/tonic/blob/b3fca19104bf001d3a3dac74221b7c9bede13cf1/tonic/src/status.rs#L617-L620).

**Comment by timvisee** [comment]
I've rebased this on the latest `dev`, to include the same changes for the new client functions we've added. I've also increased the maximum timeout to 24 hours as we'll use this high timeout value for shard snapshot transfer.

Since I've already approved this I'll merge this when CI is :green_circle:. I'm blocked by this for changes in <https://github.com/qdrant/qdrant/issues/2432>.

**Pull request merged**
